### PR TITLE
Fix broken add_subdirectory on github

### PR DESCRIPTION
**Pull request by ktf**
- Make all non core modules optional.

### CODE REVIEW

**Comment by ktf** [comment]
+1

**Comment by ktf** [comment]
+1

**Comment by dberzano** [comment]
+1

**Comment by dberzano** [comment]
+1

File: CMakeLists.txt [hunk 1]
```diff
@@ -105,6 +105,8 @@ include(cmake/GenParFile.cmake)
 set(ALILIBSTESTED "" CACHE INTERNAL "ALILIBSTESTED" FORCE)
 include(AddLibraryTested)
 
+include(cmake/modules/MacroOptionalAddSubdirectory.cmake)
```
Comment (dberzano) [inline CMakeLists.txt]: I think `cmake/modules` is the default include path, can you check?

File: EMCAL/CMakeLists.txt [hunk 2]
```diff
@@ -27,7 +27,9 @@ endif(DA)
 
 # Install the macros, docs
 install(DIRECTORY macros DESTINATION EMCAL)
-install(DIRECTORY doc DESTINATION EMCAL)
 install(DIRECTORY mapping DESTINATION EMCAL)
+if(EXISTS EMCAL/doc)
+install(DIRECTORY doc DESTINATION EMCAL)
```
Comment (dberzano) [inline EMCAL/CMakeLists.txt]: Could you indent? Thanks!

File: EVGEN/CMakeLists.txt [hunk 3]
```diff
@@ -135,6 +135,8 @@ install(TARGETS ${MODULE}
   ARCHIVE DESTINATION lib
   LIBRARY DESTINATION lib)
 install(FILES ${HDRS} DESTINATION include)
+if(EXISTS EVGEN/dataCorrHF)
 install(DIRECTORY dataCorrHF DESTINATION EVGEN)
```
Comment (dberzano) [inline EVGEN/CMakeLists.txt]: Indent

File: ITSMFT/COMMONITSMFT/CMakeLists.txt [hunk 4]
```diff
@@ -24,7 +24,9 @@ add_subdirectory(COMMONITSMFTBase)
 add_subdirectory(COMMONITSMFTSim)
 add_subdirectory(COMMONITSMFTRec)
 
+if(EXISTS ITSMFT/COMMONITSMFT/misc)
 install(DIRECTORY misc DESTINATION ITSMFT/COMMONITSMFT)
```
Comment (dberzano) [inline ITSMFT/COMMONITSMFT/CMakeLists.txt]: Indent

File: ITSMFT/ITS/CMakeLists.txt [hunk 5]
```diff
@@ -19,6 +19,10 @@ add_subdirectory(ITSUpgradeRec)
 add_subdirectory(ITSUpgradeSim)
 
 # Macros
-install(DIRECTORY macros misc FT1 FT2 itsuTestBench DESTINATION ITSMFT/ITS)
+install(DIRECTORY macros FT1 FT2 itsuTestBench DESTINATION ITSMFT/ITS)
+
+if(EXISTS ITSMFT/ITS/misc)
+install(DIRECTORY misc DESTINATION ITSMFT/ITS)
```
Comment (dberzano) [inline ITSMFT/ITS/CMakeLists.txt]: Indent

File: LHAPDF/CMakeLists.txt [hunk 6]
```diff
@@ -20,6 +20,8 @@ add_subdirectory(lhapdf5.5.1)
 # liblhapdf-5.9.1
 add_subdirectory(lhapdf-5.9.1)
 
+if(EXISTS "LHAPDF/PDFsets")
 install(DIRECTORY PDFsets DESTINATION LHAPDF)
```
Comment (dberzano) [inline LHAPDF/CMakeLists.txt]: Again :-)

File: PHOS/CMakeLists.txt [hunk 7]
```diff
@@ -41,6 +41,8 @@ set(DATFILES
     Parameters.dat
    )
 
+if(EXISTS PHOS/PCA8pa15_0.5-100.root)
 install(FILES ${ROOTFILES} ${DATFILES} DESTINATION PHOS)
```
Comment (dberzano) [inline PHOS/CMakeLists.txt]: Yet another

File: TOF/CMakeLists.txt [hunk 8]
```diff
@@ -22,7 +22,11 @@ if(DA)
 endif(DA)
 
 # Install the data folder
-install (DIRECTORY macros data DESTINATION TOF)
+install (DIRECTORY macros DESTINATION TOF)
+if(EXISTS TOF/data)
+install (DIRECTORY data DESTINATION TOF)
```
Comment (dberzano) [inline TOF/CMakeLists.txt]: You know haha

File: TPC/CMakeLists.txt [hunk 9]
```diff
@@ -44,4 +44,6 @@ set(DATFILES
     conn_oroc.dat
    )
 
+if (EXISTS TPC/AliTPCprf2dGEM.root)
 install(FILES ${ROOTFILES} ${DATFILES} DESTINATION TPC)
```
Comment (dberzano) [inline TPC/CMakeLists.txt]: Indent

File: TPC/TPCcalib/CMakeLists.txt [hunk 10]
```diff
@@ -99,6 +99,8 @@ install(TARGETS ${MODULE}
         LIBRARY DESTINATION lib)
 
 # Calib maps
+if(EXISTS TPC/TPCcalib/maps)
 install(DIRECTORY maps DESTINATION TPC/${MODULE})
```
Comment (dberzano) [inline TPC/TPCcalib/CMakeLists.txt]: Indent

File: TRD/CMakeLists.txt [hunk 11]
```diff
@@ -25,9 +25,14 @@ if(DA)
 endif(DA)
 
 # Install macros
-install(DIRECTORY macros data DESTINATION TRD)
+install(DIRECTORY macros DESTINATION TRD)
+if (EXISTS TRD/data)
+install(DIRECTORY data DESTINATION TRD)
```
Comment (dberzano) [inline TRD/CMakeLists.txt]: Indent

File: TRD/CMakeLists.txt [hunk 12]
```diff
@@ -25,9 +25,14 @@ if(DA)
...
+endif()
 
 # Install doc
+if (EXISTS TRD/doc)
 install(DIRECTORY doc DESTINATION TRD)
```
Comment (dberzano) [inline TRD/CMakeLists.txt]: Indent

File: ZDC/CMakeLists.txt [hunk 13]
```diff
@@ -22,6 +22,7 @@ if(DA)
 endif(DA)
 
 install(DIRECTORY macros DESTINATION ZDC)
+if (EXISTS ZDC/GlauberMCDist.root)
 install(FILES
```
Comment (dberzano) [inline ZDC/CMakeLists.txt]: Indent

File: CMakeLists.txt (hunk 1 above)
Comment (ktf) [inline CMakeLists.txt]: No, it's not.

File: EMCAL/CMakeLists.txt (hunk 2 above)
Comment (ktf) [inline EMCAL/CMakeLists.txt]: Yes, where is the CMake style guide? ;-) 2 or 4 spaces? I assume 4, right?

**Comment by ktf** [comment]
I think I fixed all the required bits.

**Comment by ktf** [comment]
Closing this since it should have been merged upstream.

**Pull request closed without merging**